src/ol_concourse/pipelines/infrastructure/learn_ai/docker_pulumi.py

```diff
@@ -0,0 +1,320 @@
+        branch="release",
+        fetch_tags=True,
+        tag_regex=r"v[0-9]\.[0-9]*\.[0-9]",  # examples v0.24.0, v0.26.3
+    )
+
+    # Used for publishing the CI containers to dockerhub
+    learn_ai_registry_ci_image = registry_image(
+        name=Identifier("learn-ai-ci-image"),
+        image_repository="mitodl/learn-ai-app-main",
+        username="((dockerhub.username))",
+        password="((dockerhub.password))",  # noqa: S106
+        tag_regex="[0-9A-Fa-f]+",  # Should only capture the CI images
+    )
+
+    # Used for publishing the RC / production containers to dockerhub
+    learn_ai_registry_rc_image = registry_image(
+        name=Identifier("learn-ai-rc-image"),
+        image_repository="mitodl/learn-ai-app-main",
+        username="((dockerhub.username))",
+        password="((dockerhub.password))",  # noqa: S106
+        tag_regex=r"v[0-9]\.[0-9]*\.[0-9]",  # examples v0.24.0, v0.26.3
+    )
+
+    # Fetches the pulumi code
+    ol_infra_repo = git_repo(
+        Identifier("ol-infra"),
+        uri="https://github.com/mitodl/ol-infrastructure",
+        branch="md/learn-ai",
+        # Purposely not monitoring paths or using this as a trigger
+    )
+
+    # This image is only used for the CI environment
+    learn_ai_main_image_build_job = Job(
+        name="build-learn-ai-image-from-main",
+        build_log_retention={"builds": 10},
+        plan=[
+            GetStep(get=learn_ai_main_repo.name, trigger=True),
+            container_build_task(
+                inputs=[Input(name=learn_ai_main_repo.name)],
+                build_parameters={
+                    "CONTEXT": learn_ai_main_repo.name,
+                },
+                build_args=[],
+            ),
+            PutStep(
+                put=learn_ai_registry_ci_image.name,
+                params={
+                    "image": "image/image.tar",
+                    "additional_tags": f"./{learn_ai_main_repo.name}/.git/short_ref",
+                },
+            ),
+        ],
+    )
+
+    # This image will be used for both the QA and production deployments
+    # hopefully it gets a meaningful tag
+    learn_ai_release_canidiate_image_build_job = Job(
+        name="build-learn-ai-image-from-release-candidate",
+        build_log_retention={"builds": 10},
+        plan=[
+            GetStep(get=learn_ai_release_candidate_repo.name, trigger=True),
+            container_build_task(
+                inputs=[Input(name=learn_ai_release_candidate_repo.name)],
+                build_parameters={
+                    "CONTEXT": learn_ai_release_candidate_repo.name,
+                },
+                build_args=[],
+            ),
+            PutStep(
+                put=learn_ai_registry_rc_image.name,
+                params={
+                    "image": "image/image.tar",
+                    "additional_tags": f"./{learn_ai_release_candidate_repo.name}/.git/ref",  # Should contain a tag if doof is doing his job
```

Comment:
I think this wants to be `.git/tag`

Reply:
https://github.com/concourse/git-resource?tab=readme-ov-file#additional-files-populated

Reply:
👍